src/ripple/core/impl/DatabaseCon.cpp

```diff
@@ -27,6 +27,61 @@
 
 namespace ripple {
 
+class CheckpointersCollection
+{
+    std::uintptr_t nextId_{0};
+    // Mutex protects the CheckpointersCollection
+    std::mutex m_;
+    // Each checkpointer is given a unique id. All the checkpointers that are
+    // part of a DatabaseCon are part of this collection. When the DatabaseCon
+    // is destroyed, its checkpointer is removed from the collection
+    std::map<std::uintptr_t, std::shared_ptr<Checkpointer>> checkpointers_;
+
+public:
+    std::shared_ptr<Checkpointer>
+    fromId(std::uintptr_t id)
+    {
+        std::lock_guard l{m_};
```

Comment:
Nit: for consistency, name this local 'lock' rather than 'l'.

Reply:
Also, consider renaming m_ to mutex_

Reply:
Fixed in `50335297d [fold] Longer variable names`